tests/test_restore_command.rs

```diff
@@ -100,3 +102,134 @@ fn test_restore() {
     R file1
     "###);
 }
+
+// Much of this test is copied from test_resolve_command
+#[test]
+fn test_restore_conflicted_merge() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(&test_env, &repo_path, "base", &[], &[("file", "base\n")]);
+    create_commit(&test_env, &repo_path, "a", &["base"], &[("file", "a\n")]);
+    create_commit(&test_env, &repo_path, "b", &["base"], &[("file", "b\n")]);
+    create_commit(&test_env, &repo_path, "conflict", &["a", "b"], &[]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @    conflict
+    ├─╮
+    o │  b
+    │ o  a
+    ├─╯
+    o  base
+    o
+    "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file")).unwrap()
+        , @r###"
+            <<<<<<<
+            %%%%%%%
+            -base
+            +a
+            +++++++
+            b
+            >>>>>>>
+            "###);
+    std::fs::write(repo_path.join("file"), "resolution").unwrap();
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["diff", "--color=always"]), 
+    @r###"
+    [38;5;3mResolved conflict in file:[39m
+    [38;5;1m   1[39m     : [38;5;1m<<<<<<<[39m
+    [38;5;1m   2[39m     : [38;5;1m%%%%%%%[39m
+    [38;5;1m   3[39m     : [38;5;1m-base[39m
+    [38;5;1m   4[39m     : [38;5;1m+a[39m
+    [38;5;1m   5[39m     : [38;5;1m+++++++[39m
+    [38;5;1m   6[39m     : [38;5;1mb[39m
+    [38;5;1m   7[39m     : [38;5;1m>>>>>>>[39m
+         [38;5;2m   1[39m: [38;5;2mresolution[39m
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["restore", "file"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Created 70d64d389989 conflict
+    Working copy now at: 70d64d389989 conflict
+    Added 0 files, modified 1 files, removed 0 files
+    "###);
+    // Minor BUG: Restored conflict is flipped
```

Comment:
@martinvonz I'm not sure what's going on here. I checked, and this seems unrelated to your recent changes: rebasing this branch onto 3dfedf5 doesn't seem to change the behavior.

Reply:
Actually, maybe if I get the parents without resolving a revision, it'll go better?

Reply:
It did, this is now fixed.